src/filter.ts

```diff
@@ -19,208 +34,283 @@ function match(pattern: any, value: any) {
+      getDevices(hass),
+      match(value),
+    ]);
+    return (entity) => {
+      const ent = entities.get(entity.entity_id);
+      if (!ent) return false;
+      const device = devices.get(ent.device_id);
+      if (!device) return false;
+      return matcher(device.manufacturer);
+    };
+  },
+  device_model: async (hass, value) => {
+    const [entities, devices, matcher] = await Promise.all([
+      getEntities(hass),
+      getDevices(hass),
+      match(value),
+    ]);
+    return (entity) => {
+      const ent = entities.get(entity.entity_id);
+      if (!ent) return false;
+      const device = devices.get(ent.device_id);
+      if (!device) return false;
+      return matcher(device.model);
+    };
+  },
+  area: async (hass, value) => {
+    const [entityAreas, areas, matcher] = await Promise.all([
+      getEntityAreas(hass),
+      getAreas(hass),
+      match(value),
+    ]);
+    return (entity) => {
+      const area = areas.get(entityAreas.get(entity.entity_id));
+      if (!area) return false;
+      return (
+        (area.name !== undefined && matcher(area.name)) ||
+        (area.area_id !== undefined && matcher(area.area_id))
+      );
+    };
+  },
+  entity_category: async (hass, value) => {
+    const [entities, matcher] = await Promise.all([
+      getEntities(hass),
+      match(value),
+    ]);
+    return (entity) => {
+      const ent = entities.get(entity.entity_id);
+      if (!ent) return false;
+      return matcher(ent.entity_category);
+    };
   },
-  last_changed: async (hass, value, entity) => {
-    if (!ago_suffix_regex.test(value)) value = value + default_ago_suffix;
+  last_changed: async (hass, value) => {
+    const [matcher] = await Promise.all([match(value)]);
+    return (entity) => {
+      if (!ago_suffix_regex.test(value)) value = value + default_ago_suffix;
 
-    return match(value, entity.last_changed);
+      return matcher(entity.last_changed);
+    };
   },
-  last_updated: async (hass, value, entity) => {
-    if (!ago_suffix_regex.test(value)) value = value + default_ago_suffix;
+  last_updated: async (hass, value) => {
+    const [matcher] = await Promise.all([match(value)]);
+    return (entity) => {
+      if (!ago_suffix_regex.test(value)) value = value + default_ago_suffix;
 
-    return match(value, entity.last_updated);
+      return matcher(entity.last_updated);
+    };
   },
-  last_triggered: async (hass, value, entity) => {
-    if (entity.attributes.last_triggered == null) return false;
-    if (!ago_suffix_regex.test(value)) value = value + default_ago_suffix;
+  last_triggered: async (hass, value) => {
+    const [matcher] = await Promise.all([match(value)]);
+    return (entity) => {
+      if (entity.attributes.last_triggered == null) return false;
+      if (!ago_suffix_regex.test(value)) value = value + default_ago_suffix;
 
-    return match(value, entity.attributes.last_triggered);
+      return matcher(entity.attributes.last_triggered);
```

Comment:
Could the check against `null` be included in the matcher instead? Or would that mess something else up?